python/cuml/cuml/svm/linear.pyx

```diff
@@ -643,22 +644,92 @@ class LinearSVM(Base, metaclass=WithReexportedParams):
         self.classes_ = None
         self.probScale_ = None
 
+    @classmethod
+    def _get_param_names(cls):
+        return super()._get_param_names() + [
+            "penalty",
+            "loss",
+            "fit_intercept",
+            "penalized_intercept",
+            "probability",
+            "max_iter",
+            "linesearch_max_iter",
+            "lbfgs_memory",
+            "C",
+            "grad_tol",
+            "change_tol",
+            "epsilon",
+            "tol"
+        ]
+
+    @classmethod
+    def _params_from_cpu(cls, model):
+        params = {
+            "C": model.C,
+            "fit_intercept": model.fit_intercept,
+            "max_iter": model.max_iter,
+            "tol": model.tol,
+        }
+
+        return params
+
+    def _params_to_cpu(self):
+        params = {
+            "C": self.C,
+            "fit_intercept": self.fit_intercept,
+            "max_iter": self.max_iter,
+            "tol": self.grad_tol,
+        }
+
+        return params
+
+    def _attrs_from_cpu(self, model):
+        coef_ = model.coef_.reshape(1, -1)
+        return {
+            "coef_": to_gpu(coef_, order="F", dtype=np.float64),
+            "intercept_": to_gpu(model.intercept_, order="F", dtype=np.float64),
+            "n_features_in_": model.n_features_in_,
+            **super()._attrs_from_cpu(model)
+        }
+
+    def _attrs_to_cpu(self, model):
+        return {
+            "coef_": to_cpu(self.coef_, order="C", dtype=np.float64),
+            "intercept_": to_cpu(self.intercept_, order="C", dtype=np.float64),
+            "n_features_in_": self.n_features_in_,
+            **super()._attrs_to_cpu(model)
+        }
+
+    def _sync_attrs_from_cpu(self, model):
+        super()._sync_attrs_from_cpu(model)
+        self.__sync_model()
+
     @property
     def n_classes_(self) -> int:
         if self.classes_ is not None:
             return self.classes_.shape[0]
         return self.model_.classes_.shape[0]
 
     def fit(self, X, y, sample_weight=None, *, convert_dtype=True) -> 'LinearSVM':
-
-        X_m, n_rows, _n_cols, self.dtype = input_to_cuml_array(X, order='F')
-        convert_to_dtype = self.dtype if convert_dtype else None
+        X_m, n_rows, self.n_features_in_, dtype = input_to_cuml_array(
+            X,
+            convert_to_dtype=(np.float32 if convert_dtype else None),
+            check_dtype=[np.float32, np.float64],
+            order='F')
+        if hasattr(X_m, "index"):
+            self.feature_names_in_ = X_m.index
```

Comment:
I pointed this out earlier in the review, but it must have been missed. Please don't do this - this is the incorrect way of setting `feature_names_in_` (I recognize this has been done elsewhere in cuml currently, but it's wrong). We'll fix this globally before the next release, but for now we shouldn't add an incorrect implementation.

```suggestion
```